### PR TITLE
rtr

### PR DESCRIPTION
**Pull request by olehmv**

### CODE REVIEW

File: src/test/scala/rtr/RetryTest.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      retries =
+        List(0.seconds, 1.seconds, 2.seconds))
+
+    retryListCount - initialTry should be(retryZeroTimes)
+  }
+  it should "retry two times when accepted result succeed from the second retry" in {
+    retry[Int](
+      block =
+        () => {
+          retryListCount = retryListCount + 1;
```
Comment (azhur) [inline src/test/scala/rtr/RetryTest.scala:47]: ; is redundant

File: src/test/scala/rtr/RetryTest.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    retryListCount - initialTry should be(retryTwoTimes)
+  }
+  it should "retry three times when accepted result never succeed and retries list has tree elements" in {
+    retry[Int](
+      block =
+        () => {
+          retryListCount = retryListCount + 1;
+          retryListCount
+        },
```
Comment (azhur) [inline src/test/scala/rtr/RetryTest.scala:63]: this block function could be extracted into a value and passed in all your tests

File: src/test/scala/rtr/RetryTest.scala [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      acceptResult =
+        res => false,
+      retries =
+        List(0.seconds, 1.seconds, 2.seconds))
+
+    retryListCount - initialTry should be(retryTreeTimes)
+  }
+
+
+}
```
Comment (azhur) [inline src/test/scala/rtr/RetryTest.scala:85]: can you add tests that assert backoffs?

**Comment by olehmv** [comment]
please explain what is backoffs

Отримати Outlook для Android<https://aka.ms/ghei36>

________________________________

**Comment by azhur** [comment]
@olehmv backoff is the time which your function waits between retries, you should assert that it waits.

File: src/test/scala/rtr/RetryTest.scala (hunk 3 above)
Comment (olehmv) [reply src/test/scala/rtr/RetryTest.scala]: it should "back off 2 second" in {
    import scala.concurrent.ExecutionContext.Implicits.global
    val r=Future[Int](retry[Int](()=>{1},res=>true,List(2.seconds)))
    Await.result(r,2.seconds) should be(1)
  }

**Pull request merged**